app/discovery/provider/consulcatalog/client.go

```diff
@@ -0,0 +1,147 @@
+package consulcatalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Client struct {
+	address string
+	client  *http.Client
+}
+
+func NewClient(address string, timeout time.Duration) *Client {
+	cl := &Client{
+		address: strings.TrimRight(address, "/"),
+	}
+
+	cl.client = &http.Client{
+		Timeout: timeout,
+	}
+
+	return cl
+}
+
+func (cl *Client) Get() ([]*ConsulService, error) {
```

Comment:
same as in somee other place - why slice of pointers? just []ConsulService will be easier